plugin/pkg/admission/podpreset/admission.go

```diff
@@ -240,20 +246,27 @@ func mergeEnv(envVars []api.EnvVar, podPresets []*settings.PodPreset) ([]api.Env
 	return mergedEnv, err
 }
 
-func mergeEnvFrom(envSources []api.EnvFromSource, podPresets []*settings.PodPreset) ([]api.EnvFromSource, error) {
+func mergeEnvFrom(envSources []api.EnvFromSource, podPresets []*settingsv1alpha1.PodPreset) ([]api.EnvFromSource, error) {
 	var mergedEnvFrom []api.EnvFromSource
 
 	mergedEnvFrom = append(mergedEnvFrom, envSources...)
 	for _, pp := range podPresets {
-		mergedEnvFrom = append(mergedEnvFrom, pp.Spec.EnvFrom...)
+		for _, envFromSource := range pp.Spec.EnvFrom {
+			internalEnvFrom := api.EnvFromSource{}
+			if err := apiscorev1.Convert_v1_EnvFromSource_To_core_EnvFromSource(&envFromSource, &internalEnvFrom, nil); err != nil {
```

Comment:
we're converting `Env`, `EnvFrom`, `Volume`, `VolumeMount` via the generated functions. can we possibly avoid this?

Reply:
> we're converting `Env`, `EnvFrom`, `Volume`, `VolumeMount` via the generated functions. can we possibly avoid this?

It's like the quota evaluation.  Not really and even though you didn't see the conversion when it was internal, you still had it when we read the data.